### DISCUSSION TITLE
Ability to save plots as images

### DISCUSSION THREAD

**Discussion by jameskuszmaul-brt** [foxglove/community#291] [Q&A]
It would be nice to have an option to export the current state of a Plot panel as an image. PNG would probably be adequate for most uses, although I know I've appreciated having SVG (or other vectorized formats) available for export in the past for inclusion in higher-quality reports.

The current work-around is to either take a screenshot manually or to export the CSV for the plot and manually generate the plot using other tooling (both of which are reasonable, but add friction).

**Comment by rgov** [comment]
There's a Chart.js [`toBase64Image()`](https://www.chartjs.org/docs/2.9.4/developers/api.html#tobase64image) API that could be used to implement this. A few years ago someone [tinkered with exporting to SVG](https://stackoverflow.com/questions/45563420/exporting-chart-js-charts-to-svg-using-canvas2svg-js), it might work.

**Comment by amacneil** [comment]
More generally, would be cool if you could save any panel (or the whole layout) as a png. Currently only image panel supports it.

**Comment by defunctzombie** [comment]
> More generally, would be cool if you could save any panel (or the whole layout) as a png. Currently only image panel supports it.

I am a little /shrug on this. I personally find it trivial to take a screenshot. I think I feel similarly about saving the plot to an image. SVG could be compelling (for the higher quality reports case), but I kinda feel like the answer there might be to fullscreen the panel, take a screenshot, and use that in a report?

**Comment by defunctzombie** [comment]
> Currently only image panel supports it.

Image panel supports it because it actually gets you the original data not a screenshot of the data. This is similar to the CSV download of the plot data.

**Comment by amacneil** [comment]
One thing to keep in mind is that taking screenshots isn't quite so trivial on windows/linux. It's possible, but can be quite frustrating.

**Comment by jameskuszmaul-brt** [comment]
Taking screenshots gets the pixels across (and is what makes this not too high a priority in my mind), but there's a real jump in quality/convenience to be able to grab images that already have perfect cropping and don't require fiddling with the system screenshot tools. There're also some other benefits if you can provide vectorized or higher-than-screen resolution export options, although that is also generally more work. Obviously if you have very high standards for your graphics, you are probably exporting the CSV and generating a bespoke plot using other tooling anyways. But I don't think that's all that common a use-case.

**Comment by rgov** [comment]
For high quality printed screenshots or presentations, don't forget you can increase the UI scale with <kbd>⌘+</kbd>/<kbd>⌘–</kbd> (or platform equivalent) to render the panel larger, then scale down or use an image editor to change the DPI.

**Comment by maxpolzin** [comment]
It is great to visualize data in the plot panel with Foxglove studio. 
However to generate vector graphics for reports, e.g. svg files, I currently have to export the plot data as csv and then plot it in python and export it to svg. It' be great to export the plot shown in Foxglove Studio directly as svg graphic.

**Reply by amacneil** [reply]
More generally being able to save any panel as an image would be great. I guess not all panels could export to svg though.

**Reply by maxpolzin** [reply]
I can "export" any panel as an image by making screenshots. Agreeably this is not the nicest solution, but it does the job. Vector graphics would allow much simpler editing of the exported graphics afterwards (particularly plots). Nevertheless, image export where vector graphics are not possible would also be nice.
